### PR TITLE
LPS-203765 Add display page support for calendar events

### PR DESCRIPTION
**Pull request by adolfopa**
# What is this trying to solve?

https://liferay.atlassian.net/browse/LPS-203765

Allow users to create display pages for calendar events.

# How can you verify that it works?

You'll need to enable the feature flag for `LPS-201853`. You can create display pages by the usual means; calendar event URLs are of the shape `/g/<calendarBookingId>`. For the moment, you'll need to get the calendar booking ID from the `CalendarBooking` table.

### CODE REVIEW

File: modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/provider/CalendarBookingInfoItemCapabilitiesProvider.java [hunk 1]
```diff
@@ -26,6 +28,10 @@ public class CalendarBookingInfoItemCapabilitiesProvider
 
 	@Override
 	public List<InfoItemCapability> getInfoItemCapabilities() {
+		if (!FeatureFlagManagerUtil.isEnabled("LPS-201853")) {
```
Comment (AliciaGarciaGarcia) [inline modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/provider/CalendarBookingInfoItemCapabilitiesProvider.java:31]: Error on console
```
2023-12-18 16:06:47.001 ERROR [SystemExecutorServiceUtil-8-com.liferay.portal.spring.extender.internal.configuration.ServiceConfigurationInitializer-com.liferay.template.service][FeatureFlagsBag:90] Feature flag LPS-201853 is not available for company 0
```

It's asking for a companyId, so **maybe** smoke test will fail

File: modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/CalendarBookingInfoItemFields.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import com.liferay.info.field.type.BooleanInfoFieldType;
+import com.liferay.info.field.type.DateInfoFieldType;
+import com.liferay.info.field.type.HTMLInfoFieldType;
+import com.liferay.info.field.type.TextInfoFieldType;
+import com.liferay.info.localized.InfoLocalizedValue;
+
+/**
+ * @author Eudaldo Alonso
+ */
+public class CalendarBookingInfoItemFields {
```
Comment (AliciaGarciaGarcia) [inline modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/CalendarBookingInfoItemFields.java:20]: OPTIONAL/QUESTION

This file was moved but seems that git does not get that way.

This is intentional?

File: modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/provider/CalendarBookingInfoItemFormProvider.java [hunk 3]
```diff
@@ -41,6 +41,18 @@ private InfoFieldSet _getBasicInformationInfoFieldSet() {
 		return InfoFieldSet.builder(
 		).infoFieldSetEntry(
 			CalendarBookingInfoItemFields.titleInfoField
+		).infoFieldSetEntry(
```
Comment (AliciaGarciaGarcia) [inline modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/provider/CalendarBookingInfoItemFormProvider.java:44]: Question: 

No alphabetical order is ok?

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]

File: modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/CalendarBookingInfoItemFields.java (hunk 2 above)
Comment (adolfopa) [reply modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/CalendarBookingInfoItemFields.java:20]: The move was intentional; the class was in the incorrect package.

File: modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/provider/CalendarBookingInfoItemFormProvider.java (hunk 3 above)
Comment (adolfopa) [reply modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/provider/CalendarBookingInfoItemFormProvider.java:44]: Yes; form providers list fields in a "user friendly" order (this order will be used when mapping fields in the page editor).

File: modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/provider/CalendarBookingInfoItemCapabilitiesProvider.java (hunk 1 above)
Comment (adolfopa) [reply modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/provider/CalendarBookingInfoItemCapabilitiesProvider.java:31]: You're right :facepalm:

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 4]
```diff
@@ -6939,6 +6939,7 @@ event-id=Event ID
 event-information=Event Information
 event-location=Event Location
 event-name=Event Name
+event-url=Event URL
```
[the remark this answers is not in the document]
Comment (adolfopa) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:6942]: You're right; but I wonder why SF didn't fail ...

File: modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/provider/CalendarBookingInfoItemCapabilitiesProvider.java (hunk 1 above)
Comment (adolfopa) [reply modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/provider/CalendarBookingInfoItemCapabilitiesProvider.java]: Changes squashed in https://github.com/liferay-lima/liferay-portal/pull/5018/commits/2407c67088fa1bb458b2865b0514b80f00ea670e

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties (hunk 4 above)
[the remark this answers is not in the document]
Comment (adolfopa) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:6942]: See https://github.com/liferay-lima/liferay-portal/pull/5018/commits/cf57dd34a456567a383c88229ad19eec09aa40d6

File: modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/CalendarBookingInfoItemFields.java (hunk 2 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/info/item/CalendarBookingInfoItemFields.java:20]: I was more to be aware that it feels more that 
deleted and created
than moved (github dixit)

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by Jeremy-Ch** [comment]
ci:test:relevant

**Comment by Jeremy-Ch** [comment]
Hi @adolfopa Could you help to take a look if this failure is related to this PR? Thanks.
https://testray.liferay.com/home/-/testray/case_results/index?p_p_state=normal&orderByCol=status_sortable&orderByType=asc&testrayBuildId=2477104684&testrayRunId=2477104741&errors=Feature
```
[beanshell] 2023-12-20 06:12:58.835 ERROR [SystemExecutorServiceUtil-2-com.liferay.portal.osgi.web.wab.extender.internal.WabFactory-BundleTrackerOpener][FeatureFlagsBag:90] Feature flag LPS-201853 is not available for company 0
[beanshell] 2023-12-20 06:13:08.960 ERROR [SystemExecutorServiceUtil-1-com.liferay.portal.spring.extender.internal.configuration.ServiceConfigurationInitializer-com.liferay.template.service][FeatureFlagsBag:90] Feature flag LPS-201853 is not available for company 0
```

**Comment by adolfopa** [comment]
@AliciaGarciaGarcia The company ID is not available at the point we need to check the FF; made it a system FF in https://github.com/liferay-lima/liferay-portal/pull/5018/commits/5e00b015e4f1661559e33ded695fd76807ec4934

**Comment by adolfopa** [comment]
ci:forward

**Comment by Jeremy-Ch** [comment]
The failures are unrelated.

**Comment by Jeremy-Ch** [comment]
ci:forward:force

**Pull request closed without merging**